src/framework/smoke_job.hpp

```diff
@@ -154,9 +154,10 @@ class SmokeJob {
     return task;
   }
 
-  static std::list<SmokeJob> createJobsFromJson(const SmokeFlags flags, bool&
+  static std::list<std::shared_ptr<SmokeJob>> createJobsFromJson(
```

Comment:
Of course - unique_ptr does not apply here.. We have multiple pointers to one object (we don't want to copy) to have shared state between lists.